src/corelibs/U2View/src/ov_msa/MultilineScrollController.cpp

```diff
@@ -177,11 +177,58 @@ int MultilineScrollController::getViewHeight() {
     return childrenScrollArea->height();
 }
 
+bool MultilineScrollController::checkBoundary() {
+    int firstBase = -1;
+    int prevFistBase = -1;
+    bool needUpdate = false;
+    int childrenCount = ui->getChildrenCount();
+    int alignmentLen = maEditor->getAlignmentLen();
+    for (int i = 0; i < childrenCount; i++) {
+        firstBase = ui->getUI(i)->getScrollController()->getFirstVisibleBase(false);
+        if (firstBase == prevFistBase) {
+            needUpdate = true;
+            break;
+        }
+        if ((firstBase + 1) >= alignmentLen) {
```

Comment:
All these braces with +1 are not needed in the patch

Reply:
fixed